### PR TITLE
Add slack pkg support

### PR DESCRIPTION
**Pull request by KimNorgaard**
See https://github.com/ansible/ansible/issues/3437 and https://github.com/ansible/ansible/pull/6364

### CODE REVIEW

File: packaging/os/slackpkg.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+
+DOCUMENTATION = '''
+---
+module: slackpkg
+short_description: Package manager for Slackware >= 12.2
+description:
+    - Manage binary packages for Slackware using 'slackpkg' which
+      is available in versions after 12.2.
+version_added: "1.6"
```
Comment (resmo) [inline packaging/os/slackpkg.py]: should be "2.0"

File: packaging/os/slackpkg.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+
+DOCUMENTATION = '''
+---
+module: slackpkg
+short_description: Package manager for Slackware >= 12.2
```
Comment (resmo) [inline packaging/os/slackpkg.py:28]: I would put the ">= 12.2" to the requirements

File: packaging/os/slackpkg.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+def update_cache(module, slackpkg_path):
+    rc, out, err = module.run_command("%s -batch=on update" % (slackpkg_path))
+    if rc != 0:
+        module.fail_json(msg="Could not update package cache")
+
+def main():
+    module = AnsibleModule(
+        argument_spec       = dict(
+            state           = dict(default="installed", choices=['installed', 'removed', 'absent', 'present', 'latest']),
+            name            = dict(aliases=["pkg"], required=True),
```
Comment (resmo) [inline packaging/os/slackpkg.py]: use `type='list'` and you can pass a comma separated string as well as a yaml list. The result would be a python list, so your code must be adjusted accordantly.

File: packaging/os/slackpkg.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        default: present
+
+    update_cache:
+        description:
+            - update the package database first
+        required: false
+        default: false
+        choices: [ true, false ]
+
+author: Kim Nørgaard
```
Comment (resmo) [inline packaging/os/slackpkg.py]: author format changed, please add your GitHub handle to the author e.g `author:  Kim Nørgaard (@KimNorgaard)"`

**Comment by KimNorgaard** [comment]
Thanks for pointing out the issues. They should be fixed now.

**Comment by resmo** [comment]
LGTM +1, handover to core review @gregdek

**Pull request merged**